server/bin/pbench-index.py

```diff
@@ -292,6 +350,16 @@ def main(options, name):
             skipped = Path(tmpdir, f"{name}.{idxctx.TS}.skipped")
             ie_filepath = Path(tmpdir, f"{name}.{idxctx.TS}.indexing-errors.json")
 
+            # We use a list object here so that when we close over this
```

Comment:
I don't understand this: why isn't a boolean enough?

Reply:
When you "close" over a variable in a function definition, the value of that variable remains as it was for each invocation of that function when that function is not called in the same scope.  Here is an example:

```
import sys

def setup_closure(value_i, value_l):
    def func_closure():
        print(f"value_i = {value_i!r}; value_l = {value_l!r}")
    return func_closure

def main(argv):
    value_i = 42
    value_l = [ 42 ]

    def func_closure():
        print(f"value_i = {value_i!r}; value_l = {value_l!r}")

    # The next two calls work as expected because the scope
    # of value_i and value_l[0] are the same.
    func_closure()
    value_i += 1
    value_l[0] += 1
    func_closure()

    cl = setup_closure(value_i, value_l)

    # The next two calls don't work as expected because the
    # scope of value_i is not the same.
    cl()
    value_i += 1
    value_l[0] += 1
    cl()

    return 0

if __name__ == "__main__":
    sys.exit(main(sys.argv))
```